docs(all): clarifies what is overridden on a per-api basis (#1882)
Redocly/redocly-cli: docs/configuration/apis.md

For every API listed in the object, you must provide the path to the OpenAPI description using the `root` property.

If `rules`, `decorators`, or `preprocessors` aren't defined for an API, root settings are used.
If `rules`, `decorators`, or `preprocessors` are defined for an API, they apply together with the root configuration.

> If per-API `rules`, `decorators`, or `preprocessors` and root settings modify the same properties, per-API `rules`, `decorators`, and `preprocessors` override root settings.

Reviewer: It's not too obvious what properties/settings are in that context. Could it be rephrased like this?

```suggestion
If the same `rules`, `decorators`, or `preprocessors` are defined on `apis` and root level, per-API `rules`, `decorators`, and `preprocessors` override root ones.
```

> For example, if you include the same `decorator` at the root level and for a specific API, but with different properties, only the properties applied to the specific API will be applied to that API.

Reviewer: > only the properties applied to the specific API will be applied to that API.

Seems like a tautology to me :). 

```suggestion
For example, if you include the same `decorator` at the root level and for a specific API, but with different properties, the API-level settings replace the root ones.
```

So if you have the following `redocly.yaml` configuration, adding `filter-in` and `plugin/change-title` at the root level and applying `plugin/change-title` with a different `title` property to the `storefront@latest` API:

```yaml
decorators:
  filter-in:
    property: x-products
    value:
      - Core
  plugin/change-title:

>     title: All APIs

Reviewer: I'd add some other parameter here, so it's more clear that it will be overriden:

```suggestion
    title: All APIs
    extraProperty: will be ignored on per-API level
```

>     property-four: 4

Reviewer: I don't really like that `decorator-one` and `property-one` namings. Can we use something closer to the real-world scenario? Probably we can borrow something [here](https://github.com/Rebilly/api-definitions/blob/main/redocly.yaml).

apis:
  override@v1:
    decorators:
      decorator-one:
        property-five: 5
```

> Only `property-five` is applied to `decorator-one` for the `override@v2` API, and both `property-three` and `property-four` are applied for `decorator-two`.

Reviewer: Did you mean `@v1`?

> The `plugin/change-title` decorator with the "Storefront APIs" `title` property is applied to the `storefront@latest` API, and the `filter-in` decorator is also applied to the `storefront@latest` API.
> 
> For all other APIs, not including the `storefront@latest` API, `filter-in` and `plugin/change-title` with the "Core" `title` property are applied.

Reviewer: ```suggestion
The `plugin/change-title` decorator with the "Storefront APIs" `title` property is applied to the `storefront@latest` API with value `Storefront APIs`, and the `filter-in` decorator is also applied to the `storefront@latest` API.

For all other APIs, not including the `storefront@latest` API, `filter-in` and `plugin/change-title` with the "Core" `title` and `extraProperty` properties are applied.
```

For every API listed in the object, you must provide the path to the OpenAPI description using the `root` property.

If `rules`, `decorators`, or `preprocessors` aren't defined for an API, root settings are used.

> If `rules`, `decorators`, or `preprocessors` are defined for an API, its first-level properties apply together with the root configuration.

Reviewer: It's not first-level properties of rules &c., but rather rules, decorators, and preprocessors are the first-level properties of a per-api configuration. 

For instance, let's define this in **redocly.yaml**:

```yaml
decorators:
  decorator-one: # <-- first level `decorators` property is a decorator (they get merged)
     prop1: 1 # <-- decorator's properties are something we don't merge but replace
     prop2: 2
  decorator-two:
     prop3: 3
     prop4: 4

apis:
   override@v1:
      decorators:
         decorator-one:
             prop: 5
```

When applied to `override@v1`, that results in the following config:

```yaml
decorators:
  decorator-one: # <-- got replaced entirely (including its props)
             prop: 5
  decorator-two: # <-- remains unchanged as it gets merged with the per-api decorators
     prop3: 3
     prop4: 4
```

The same goes for `theme` first-level properties (which is deprecated already, the they should become just root-level properties).

So it's a bit trickier than that 😅.
